### PR TITLE
[AutoMM] Advanced techniques for tabular MLPs

### PR DESCRIPTION
**Pull request by yiqings**
Some advanced techniques for tabular MLPs
1.  Support numerical feature embeddings [1] in `numerical_mlp` (previously, we have already supported it in `numerical_transformer`). 
2. Support `ghost batch normalization` in `numerical_mlp` and `categorical_mlp`. Usage: `model.numerical_mlp.normalization: ghost_batch_norm`. This techniques is shown in [2] to be effective. 
(Note that the `Leaky gates` introduced in [2] is a special case of `NumericalEmbedding` by setting `model.numerical_transformer.embedding_arch: ['linear', 'leaky_relu']`.)

[1] On Embeddings for Numerical Features in Tabular Deep Learning, https://arxiv.org/abs/2203.05556
[2] Simple Modifications to Improve Tabular Neural Networks: https://arxiv.org/pdf/2108.03214

### CODE REVIEW

**Comment by sxjscience** [comment]
Do we have benchmark numbers?

**Comment by yiqings** [comment]
> Do we have benchmark numbers?

Not really. Currently we only have results on CA. The experiments on 10 other datasets used in examples are still on going. We found it takes a bit longer time to train MLPs than FT_Transformers under the same optimization configurations.

**Comment by sxjscience** [comment]
@yiqings I'm okay to merge this in if it won't break our current code.

**Comment by szha** [comment]
Job PR-1865-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1865/1/index.html

**Comment by szha** [comment]
Job PR-1865-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1865/2/index.html

**Review by sxjscience — APPROVED** [review]
LGTM

**Comment by szha** [comment]
Job PR-1865-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1865/4/index.html
